From d782595e1b84190f163d0b138dae10487b538738 Mon Sep 17 00:00:00 2001
From: "Christensen, Taylor" <taylor.christensen@intel.com>
Date: Tue, 3 Nov 2015 16:01:29 -0800
Subject: [PATCH 224/441] PCIe: PME Poll Toggle Code

This patch fine tunes the PME poll capabilities of
the device to disable when PM is enabled allowing the
port to suspend properly
---
 drivers/pci/pcie/portdrv_pci.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- a/drivers/pci/pcie/portdrv_pci.c
+++ b/drivers/pci/pcie/portdrv_pci.c
@@ -154,9 +154,13 @@ static int pcie_port_runtime_idle(struct
 	 */
 	pci_walk_bus(pdev->subordinate, pci_dev_pme_poll, &pme_poll);
 	/* Delay for a short while to prevent too frequent suspend/resume */
-	/* if (!pme_poll) */ /* Disabling pme_poll check for the sake of PM  *
-	                      * on supported devices                         */
-	pm_schedule_suspend(dev, 10);
+	if (atomic_read(&dev->power.usage_count) > 0) {
+		if (!pme_poll)
+			pm_schedule_suspend(dev, 10);
+	}
+	else{
+		pm_schedule_suspend(dev, 10);
+	}
 	return -EBUSY;
 }
 #else
